client/state/data-layer/wpcom/sites/plugins/new/index.js

```diff
@@ -21,6 +28,38 @@ export const uploadPlugin = ( { dispatch }, action ) => {
 	}, action ) );
 };
 
+const showSuccessNotice = ( dispatch, { name } ) => {
+	dispatch( successNotice(
+		translate( 'Successfully uploaded plugin %(name)s.', {
+			args: { name }
+		} ),
+		{ duration: 5000 }
+	) );
+};
+
+const showErrorNotice = ( dispatch, error ) => {
+	const knownErrors = {
+		exists: translate( 'Upload problem: Plugin already installed on site.' ),
+		'too large': translate( 'Upload problem: Plugin zip must be smaller than 10MB.' ),
+		incompatible: translate( 'Upload problem: Incompatible plugin.' ),
+		unsupported_mime_type: translate( 'Upload problem: Not a valid zip file.' ),
+	};
+	const errorString = toLower( error.error + error.message );
+	const knownError = find( knownErrors, ( v, key ) => includes( errorString, key ) );
```

Comment:
The errors come from various different places—some from the jetpack plugin, others from wpcom—so there is not a consistency of error codes. Smushing the code and message together and doing a fairly fuzzy match allows us to match all the known errors with one lookup. 

Also, with themes we found we needed to tweak things as new error reports came in.